### PR TITLE
feat(chowly-client): add initial chowly integration

### PR DESCRIPTION
**Pull request by KanoniPan**

### CODE REVIEW

File: bin/prepublish.js
Comment (strdr4605) [inline bin/prepublish.js]:
```suggestion
writeFileSync('./dist/package.json', JSON.stringify(distPackage, null, 2));
```

File: package.json
Comment (strdr4605) [inline package.json]: Maybe we should pick another name and also use the namespace.
```suggestion
    "name": "@goparrot/chowly-api-client",
```

File: src/client/chowly-client.api.ts
Comment (strdr4605) [inline src/client/chowly-client.api.ts]: What does `toos` mean?

Comment (KanoniPan) [reply src/client/chowly-client.api.ts]: its chowly related prefix

File: package.json
Comment (KanoniPan) [reply package.json]: was inspired by google, they have almost the same naming for this kind of features

File: src/client/chowly-client.api.ts
Comment (strdr4605) [inline src/client/chowly-client.api.ts]: Maybe it makes sense to rename class as `ApiClient` as it already in **chowly** package/namespace?

File: package.json
Comment (strdr4605) [reply package.json]: Then to follow the convention of @goparrot packages, we can use `@goparrot/chowly`.

File: src/client/chowly-client.api.ts
Comment (KanoniPan) [reply src/client/chowly-client.api.ts]: btw inside **square-connect-plus** we have SquareClient, i think we need one-two more opinions about that

File: .github/ISSUE_TEMPLATE/1-bug-report.md
Comment (strdr4605) [inline .github/ISSUE_TEMPLATE/1-bug-report.md:1]: Too many changes for one PR. 

Issue/PR templates and workflows could be done in 2 separate PRs

File: src/client/chowly-client.api.ts
Comment (strdr4605) [reply src/client/chowly-client.api.ts]: I was looking at the original `square-connect`

```
var SquareConnect = require('square-connect');
var defaultClient = SquareConnect.ApiClient.instance;
```

and looks like they use `ApiClient`

Comment (KanoniPan) [reply src/client/chowly-client.api.ts]: ok, i see, but they have `SquareConnect` so user know that it comes from Square, anyway lets wait for one more review and if we get the most votes on `ApiClient` i will change it ;)

File: .nycrc.json
Comment (dragosh1011) [inline .nycrc.json]: is this the current coverage value?

File: package.json
Comment (dragosh1011) [inline package.json:61]: add `axios` and `class-transformer` to `peerDependencies` specifying the minimum acceptable versions

File: .nycrc.json
Comment (KanoniPan) [reply .nycrc.json]: current coverage is 100%

File: src/client/chowly-client.api.ts
Comment (dragosh1011) [inline src/client/chowly-client.api.ts]: remove `.api` prefix or rename cholwy-api-client

File: src/client/chowly-client.api.ts
Comment (dragosh1011) [reply src/client/chowly-client.api.ts]: I am for `ChowlyApiClient`

File: src/client/chowly-client.api.ts
Comment (dragosh1011) [inline src/client/chowly-client.api.ts]: remove `console.log`

File: src/client/chowly-client.api.ts
Comment (dragosh1011) [inline src/client/chowly-client.api.ts]: why you need `Math.random()`?

**Review by dragosh1011 — CHANGES_REQUESTED** [review]

File: .nycrc.json
Comment (dragosh1011) [reply .nycrc.json]: update values in this file to 100% then

File: src/client/chowly-client.api.ts
Comment (KanoniPan) [reply src/client/chowly-client.api.ts]: some sort of delay, still thinking how to implement this correctly, or to use exponentialDelay from google, or to make something like this with given milliseconds + some kind of random

**Review by dragosh1011 — APPROVED** [review]

File: bin/prepublish.js
Comment (dragosh1011) [inline bin/prepublish.js:16]: add `delete distPackage.jest`

File: src/interface/chowly.interface.ts
Comment (dragosh1011) [inline src/interface/chowly.interface.ts]: separate configuration interface and menu interfaces in different files

File: src/interface/chowly.interface.ts
Comment (dragosh1011) [inline src/interface/chowly.interface.ts]: `image_url?: string;`

File: package.json
Comment (strdr4605) [inline package.json]: Jest settings are getting big.
Maybe is better to extract in `jest.config.js`?

Comment (KanoniPan) [reply package.json]: sure i will extract it

File: README.md
Comment (ddimitrioglo) [inline README.md]: Do not use (async () => {})() if you are not using await inside it

File: package.json
Comment (ddimitrioglo) [inline package.json]: in `.nvmrc` you have set v12, I think you have to consolidate

File: package.json
Comment (ddimitrioglo) [inline package.json]: Everything you have done during working hours - is GoParrot's

File: package.json
Comment (ddimitrioglo) [inline package.json]: 13 is not LTS, better use ^12

File: src/client/chowly-api-client.ts
Comment (ddimitrioglo) [inline src/client/chowly-api-client.ts:10]: If I'm not mistaken Axios has special option for this `prefix: '/api/toos/v2'` I guess

File: src/interface/configuration.interface.ts
Comment (ddimitrioglo) [inline src/interface/configuration.interface.ts]: maybe `IClientConfig`?

File: src/utils/common.ts
Comment (ddimitrioglo) [inline src/utils/common.ts]: leave only `@throws`

Comment (KanoniPan) [reply src/utils/common.ts]: why leave only that? better then to remove this block

File: src/client/chowly-api-client.ts
Comment (KanoniPan) [reply src/client/chowly-api-client.ts:10]: axios interface doesn't have prefix, i tried some varianties, but it seems like it doesn't work, the only thing that worked for me was url, but it overrides /menu path

**Review by ddimitrioglo — APPROVED** [review]

**Pull request merged**
